yarn-project/blob-sink/src/client/http.ts

```diff
@@ -296,6 +401,41 @@ export class HttpBlobSinkClient implements BlobSinkClientInterface {
     }
   }
 
+  /**
+   * Try all filestores once (shuffled for load distribution).
+   * @param blobHashes - The blob hashes to fetch
+   * @param fillResults - Callback to fill in results
+   * @param ctx - Logging context
+   */
+  private async tryFileStores(
+    blobHashes: Buffer[],
+    fillResults: (blobs: BlobJson[]) => BlobWithIndex[],
+    ctx: { blockHash: string; blobHashes: string[]; indices?: number[] },
+  ): Promise<void> {
+    // Shuffle clients for load distribution
+    const shuffledClients = shuffleArray([...this.fileStoreClients]);
+
+    for (const client of shuffledClients) {
+      try {
+        const blobHashStrings = blobHashes.map(h => `0x${h.toString('hex')}`);
+        this.log.trace(`Attempting to get ${blobHashStrings.length} blobs from filestore`, {
+          url: client.getBaseUrl(),
+          ...ctx,
+        });
+        const blobs = await client.getBlobsByHashes(blobHashStrings);
+        if (blobs.length > 0) {
+          const result = fillResults(blobs);
+          this.log.debug(`Got ${blobs.length} blobs from filestore (total: ${result.length}/${blobHashes.length})`, {
+            url: client.getBaseUrl(),
+            ...ctx,
+          });
+        }
+      } catch (err) {
+        this.log.warn(`Failed to fetch from filestore: ${err}`, { url: client.getBaseUrl() });
+      }
+    }
```

Comment:
I may be missing something, but aren't we missing an early return here once we've collected all blobs? Also, if we are looking for two blobs and we got one from the first filestore, I think we keep requesting the original two from the next ones, since `blobHashes` does not get updated.